ietf-layer0-types-ext.yang

```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
+  identity QAM16 {
+    base modulation;
+    description
+      "QAM16 (Quadrature Amplitude Modulation)";
+  }
+  identity DP-QAM8 {
+    base modulation;
+    description
+      "DP-QAM8 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+  identity DC-DP-QAM8 {
+    base modulation;
+    description
+    "DC DP-QAM8 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+  identity DP-QAM16 {
+    base modulation;
+    description
+      "DP-QAM16 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+  identity DC-DP-QAM16 {
+    base modulation;
+    description
+    "DC DP-QAM16 (Dual Polarization Quadrature 
+    Amplitude Modulation)";
+  } 
+
+  identity FEC {
+    description
+      "Enumeration that defines the type of
+       Forward Error Correction";
+  }
+  identity reed-solomon {
+    base FEC;
+      description
+  "Reed-Solomon error correction";
+  }
+  identity hamming-code {
+    base FEC;
+      description
+  "Hamming Code error correction";
+  }
+  identity golay {
+    base FEC;
+      description "Golay error correction";
+  } 
+
+  identity line-coding {
+    description
+      "base line-coding class";
+    reference 
+      "ITU-T G.698.2-201811 section 7"   
+  }  
+
+  identity line-coding-NRZ-2p5G {
+    base line-coding;
+    description 
+      "ITU-T G.698.2-201811 section 7 table 8-1 "   
+  }
+
+  identity line-coding-NRZ-OTU1 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-2 "   
+
+  }
+
+  identity line-coding-NRZ-10G {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-3/8-5 "   
+
+  }
+
+  identity line-coding-NRZ-OTU2 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-4/8-6 "   
+
+  }
+
+// typedefs                
+
+  typedef operational-mode {
+    type string;
+    description
+      "Vendor-specific mode that guarantees 
+      interoperability.";
+    
+  }
+
+// temporary defined here for disalignment with 
+//ietf-layer0-types module
+  typedef standard-mode {
+    type string;
+    description
+      "ITU-T G.698.2 standard mode that guarantees 
+       interoperability.
+       It must be an string with the following format:
+       B-DScW-ytz(v) where all these attributes 
+       are conformant
+       to the ITU-T recomendation";
+    reference "ITU-T G.698.2 (11/2018)";
+  }
+
+  typedef frequency-thz {
+      type decimal64 {
+        fraction-digits 6;
+      }
+      units THz;
+      description
+        "The DWDM frequency in THz, e.g., 193.112500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+    }
+
+// GROUPINGS  
+
+/* This grouping represent the list of capabilities attributes not under operational-mode */
+
+
+  grouping common-tranceiver-capabilities {
+    description "Capability of an optical tranceiver";
+
+ 
+/* transmitter tuning range (f_tx-min, f_tx-max) */

```

Comment:
G.698.2 application codes also defines the minimum and maximum central frequency to be supported (see clause 7.2.2 of G.698.2)

Reply:
**2020-08-13 CCAMP call**

This range is needed for operational modes

@sergiobelotti Move these attributes to a new grouping which is used by both operational mode and explicit mode and not used by the ITU-T application code